### PR TITLE
Add serialization and deserialization for sharing Url

### PR DESCRIPTION
**Pull request by guergana**
first commit: serializer
second commit: deserializer

### CODE REVIEW

File: src/serialization/QuerySerializer.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import QueryRepresentation from '@/sparql/QueryRepresentation';
+
+export default class QuerySerializer {
+	public serialize( query: QueryRepresentation ): object {
```
Comment (micgro42) [inline src/serialization/QuerySerializer.ts]: That serializer should probably return a string, not an object.

Comment (micgro42) [inline src/serialization/QuerySerializer.ts]: I'm sorry, but you can't start with a `QueryRepresentation`. The `query` getter (rightly) assumes an already valid Query Builder, but we want to be able to generate this link even with just a partially filled query builder.
So the input should be the root state itself. That being said, it is probably enough to serialize the same information that will end up in the `QueryRepresentation`, i.e. no need to serialize errors or labels or the `conditionId`.

Comment (guergana) [reply src/serialization/QuerySerializer.ts]: Ok! Makes sense too.

Comment (guergana) [reply src/serialization/QuerySerializer.ts]: Adressed

**Review by Ladsgroup — APPROVED** [review]

**Pull request merged**